### ISSUE TITLE
Pathfinding Behavior

### ISSUE THREAD

**Issue by nbougalis** [XRPLF/rippled#3371]
A potential issue with the `path_find` and `ripple_path_find` when using both `destination_amount` and `send_max` was reported by Wietse Wind. 

```
{
  command: 'path_find',
  subcommand: 'create',
  source_account: 'rPdvC6ccq8hCdPKSPJkPmyZ4Mi1oG2FFkT',
  destination_account: 'rPdvC6ccq8hCdPKSPJkPmyZ4Mi1oG2FFkT',
  send_max: { 
    value: '10',
    currency: 'USD',
    issuer: 'rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B'
  },
  destination_amount: { 
    value: '-1',
    currency: 'EUR',
    issuer: 'rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq'
  }
}
```

This works like a charm. Docs on `send_max` and `destination_amount` and the `-1` value notation:

![image (2)](https://user-images.githubusercontent.com/5085415/80128194-2bf87a00-854a-11ea-822f-5727e330e7d2.png)

![image (3)](https://user-images.githubusercontent.com/5085415/80128205-30249780-854a-11ea-8499-1e9055582c55.png)

So, not using source_currencies, am using send_max, and I specify the -1 value for destination amount, so it’ll just go for send_max amount. » Works like a charm!

![image (4)](https://user-images.githubusercontent.com/5085415/80128311-52b6b080-854a-11ea-9838-caa5b3ac7e33.png)

Now I do the same thing, but with “20000000” (20 XRP) in send_max.

```
{
  "command": "path_find",
  "subcommand": "create",
  "source_account": "rPdvC6ccq8hCdPKSPJkPmyZ4Mi1oG2FFkT",
  "destination_account": "rPdvC6ccq8hCdPKSPJkPmyZ4Mi1oG2FFkT",
  "send_max": "20000000",
  "destination_amount": { 
    "value": "-1",
    "currency": "EUR",
    "issuer": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq"
  }
}
```

Works like a charm :slightly_smiling_face:

![image (8)](https://user-images.githubusercontent.com/5085415/80128733-ef794e00-854a-11ea-96a2-f7d65e846082.png)

So now, last but not least, the other way around.
Send max = IOU
destination_amount = -1.
Docs say that should work.

![image (6)](https://user-images.githubusercontent.com/5085415/80128488-98737900-854a-11ea-9af6-142da314dad5.png)

**Except it doesn’t!** :wink:

So IOU to IOU works, XRP to IOU works, but IOU to XRP doesn’t, although the docs explicitly mention I can add XRP value notation (just string in drops)

```
{
  "command": "ripple_path_find",
  "source_account": "rPdvC6ccq8hCdPKSPJkPmyZ4Mi1oG2FFkT",
  "destination_account": "rPdvC6ccq8hCdPKSPJkPmyZ4Mi1oG2FFkT",
  "send_max": { 
    "value": "10",
    "currency": "EUR",
    "issuer": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq"
  },
  "destination_amount": "-1"
}
```

![image (7)](https://user-images.githubusercontent.com/5085415/80128598-c0fb7300-854a-11ea-9673-67bd6971e72d.png)

Always empty, no paths, no alternatives, for any send_max IOU

Confirmed this happens both for `path_find` and `ripple_path_find` btw.

**Comment by WietseWind** [comment]
Maybe this is by design (?) but the reason for informing/reporting (to Nik) is that it actually does work as expected for 'one way', and the docs mention the -1 notation could be used for XRP values.

**Comment by seelabs** [comment]
Thanks @WietseWind This is a bug. I'll make sure a fix gets merged before 1.6 is released.

**Comment by WietseWind** [comment]
> Thanks @WietseWind This is a bug. I'll make sure a fix gets merged before 1.6 is released.

Hey @seelabs, do you know if this made into 1.6.0 (just released)?

**Comment by seelabs** [comment]
@WietseWind No, the patch to fix this has not been merged. Unfortunately, I added this fix to related patch that is problematic for other reasons and didn't make 1.6. When that patch got blocked I should have unbundled the pathfinding fix from the rest of the patch before 1.6 was released. I apologize, as I did say the patch would make 1.6. I'll update this issue once the fix is merged into develop.

**Comment by WietseWind** [comment]
Hey @seelabs - just a friendly reminder/check to see if this is going to be included in the next version ;)

**Comment by seelabs** [comment]
@WietseWind Yes, this is still on track to be included in 1.7.

**Comment by WietseWind** [comment]
@seelabs Just to confirm: still to be included in 1.7? As I don't see it on the 1.7 project board?

**Comment by seelabs** [comment]
@WietseWind Yes, this will be included in 1.7.

**Comment by seelabs** [comment]
@WietseWind The fix is merged into 1.7 with this patch: 36be4856f (`Fix path finding for XRP dst and sendmax`). To test it also turn on the `fixSTAmountCanonicalize` amendment.

**Comment by WietseWind** [comment]
Thank you @seelabs!

**Comment by intelliot** [comment]
Fixed by #3454
